pkg/odo/cli/dev/dev.go

```diff
@@ -114,10 +125,43 @@ func (o *DevOptions) Run() error {
 	var path = filepath.Dir(o.Context.EnvSpecificInfo.GetDevfilePath())
 
 	fmt.Fprintf(o.out, "Starting your application on cluster in developer mode\n")
-	err = o.clientset.DevClient.Start(o.Context.EnvSpecificInfo.GetDevfileObj(), platformContext, o.ignorePaths, path, os.Stdout)
+	d := DevHandler{}
+	err = o.clientset.DevClient.Start(o.Context.EnvSpecificInfo.GetDevfileObj(), platformContext, o.ignorePaths, path, os.Stdout, &d)
 	return err
 }
 
+func (o *DevHandler) RegenerateAdapterAndPush(pushParams common.PushParameters, watchParams watch.WatchParameters) error {
+	var adapter common.ComponentAdapter
+
+	adapter, err := regenerateComponentAdapterFromWatchParams(watchParams)
+	if err != nil {
+		return fmt.Errorf("unable to generate component from watch parameters: %w", err)
+	}
+
+	err = adapter.Push(pushParams)
+	if err != nil {
+		return fmt.Errorf("watch command was unable to push component: %w", err)
+	}
+
+	return nil
+}
+
+func regenerateComponentAdapterFromWatchParams(parameters watch.WatchParameters) (common.ComponentAdapter, error) {
+
+	// Parse devfile and validate. Path is hard coded because odo expects devfile.yaml to be present in the pwd/cwd.
+	devObj, err := devfile.ParseAndValidateFromFile("./devfile.yaml")
```

Comment:
Please use functions from `location` package to take care of `.devfile.yaml` file

Reply:
d02f77a